File: LeetCode/2433/golang/lc2433.go
package lc2433

// Runtime complexity: O(n)
// Auxiliary space: O(1) (O(n) if you have to count the answer array)
// Subjective level: easy
// Solved on: 2026-01-31
func findArray(pref []int) []int {
	L := len(pref)
	ans := make([]int, L)
	cumulativeXor := 0
	for i := range pref {
		ans[i] = cumulativeXor ^ pref[i]
		cumulativeXor ^= ans[i]
	}
	return ans
}

File: LeetCode/2433/cpp/lc2433.cpp
#include <cassert>
#include <vector>

class Solution {
public:
  std::vector<int> findArray(std::vector<int>& pref) {
    std::vector<int> ans(pref.size(), 0);
    int cumulative_xor = 0;
    for (int i = 0; i < pref.size(); i++) {
      ans[i] = cumulative_xor ^ pref[i];
      cumulative_xor ^= ans[i];
    }
    return ans;
  };
};

int main() {
  Solution s;

  std::vector<int> example1 = {5, 2, 0, 3, 1};
  std::vector<int> expected_answer1 = {5, 7, 2, 3, 2};
  assert(s.findArray(example1) == expected_answer1);

  std::vector<int> example2 = {13};
  std::vector<int> expected_answer2 = {13};
  assert(s.findArray(example2) == expected_answer2);
}
